include/seqan3/core/algorithm/parameter_pack.hpp

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <utility>
+
+#include <seqan3/std/concepts>
+#include <seqan3/std/type_traits>
+
+namespace seqan3::detail
+{
+
+/*!\brief Applies a function to a range of arguments.
+ * \ingroup algorithm
+ *
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ * \tparam ...args_t        The parameter pack of the arguments. (Note that each argument type can be different)
+ *
+ * \param  fn               The function to call on every argument.
+ * \param  args             The arguments as parameter pack.
+ *
+ * \details This function behaves like std::for_each but on parameter packs.
+ * The invocation will be done without any loop.
```

Comment:
Add a note, that the order of fn and the `range` of values is different due to the parameter pack expansion.